File: backend/src/main/java/com/admission/entity/Student.java
package com.admission.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("student")
public class Student {
    @TableId(value = "exam_number", type = IdType.INPUT)
    private String examNumber;

    private String name;

    private String gender;

    private LocalDate birthDate;

    private String className;

    private String nativePlace;

    private BigDecimal gaokaoScore;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

File: frontend/src/api/student.ts
import request from '../utils/request'

export interface Student {
  examNumber: string
  name: string
  gender: string
  birthDate: string
  className: string
  nativePlace: string
  gaokaoScore: number
  createTime?: string
  updateTime?: string
}

export interface StudentSearchParams {
  name?: string
  className?: string
  minScore?: number
  maxScore?: number
}

// 获取学生列表（分页）
export const getStudents = (params: {
  current: number
  size: number
} & StudentSearchParams = { current: 1, size: 10 }) => {
  return request.get('/students', { params })
}

// 添加学生
export const addStudent = (data: Student) => {
  return request.post('/students', data)
}

// 更新学生
export const updateStudent = (examNumber: string, data: Partial<Student>) => {
  return request.put(`/students/${examNumber}`, data)
}

// 删除学生
export const deleteStudent = (examNumber: string) => {
  return request.delete(`/students/${examNumber}`)
}

// 搜索学生
export const searchStudents = (params: StudentSearchParams) => {
  return request.get('/students/search', { params })
}